airbyte-server/src/main/java/io/airbyte/server/converters/WorkspaceWebhookConfigsConverter.java

```diff
@@ -10,42 +10,58 @@
 import io.airbyte.commons.json.Jsons;
 import io.airbyte.config.WebhookConfig;
 import io.airbyte.config.WebhookOperationConfigs;
+import java.util.ArrayList;
 import java.util.Collections;
+import java.util.Iterator;
 import java.util.List;
 import java.util.UUID;
+import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
+// NOTE: we suppress this warning because PMD thinks it can be a foreach loop in toApiReads but the
+// compiler disagrees.
+@SuppressWarnings("PMD.ForLoopCanBeForeach")
 public class WorkspaceWebhookConfigsConverter {
 
-  public static JsonNode toPersistenceWrite(List<WebhookConfigWrite> apiWebhookConfigs) {
+  public static JsonNode toPersistenceWrite(List<WebhookConfigWrite> apiWebhookConfigs, Supplier<UUID> uuidSupplier) {
     if (apiWebhookConfigs == null) {
       return Jsons.emptyObject();
     }
 
     final WebhookOperationConfigs configs = new WebhookOperationConfigs()
-        .withWebhookConfigs(apiWebhookConfigs.stream().map(WorkspaceWebhookConfigsConverter::toPersistenceConfig).collect(Collectors.toList()));
+        .withWebhookConfigs(apiWebhookConfigs.stream().map((item) -> toPersistenceConfig(uuidSupplier, item)).collect(Collectors.toList()));
 
     return Jsons.jsonNode(configs);
   }
 
-  public static List<WebhookConfigRead> toApiReads(List<WebhookConfig> persistenceConfig) {
-    if (persistenceConfig.isEmpty()) {
+  public static List<WebhookConfigRead> toApiReads(final JsonNode persistenceConfig) {
+    if (persistenceConfig == null) {
       return Collections.emptyList();
     }
-    return persistenceConfig.stream().map(WorkspaceWebhookConfigsConverter::toApiRead).collect(Collectors.toList());
+
+    // NOTE: we deserialize it "by hand" because the secrets aren't hydrated, so we can't deserialize it
+    // into the usual shape.
+    // TODO(mfsiega-airbyte): find a cleaner way to handle this situation.
+    List<WebhookConfigRead> configReads = new ArrayList<>();
+    final JsonNode configArray = persistenceConfig.findPath("webhookConfigs");
+    for (Iterator<JsonNode> it = configArray.elements(); it.hasNext();) {
+      JsonNode webhookConfig = it.next();
+      configReads.add(toApiRead(webhookConfig));
+    }
```

Comment:
I think this would be easier to read if replaced with a while loop. Something like:

```
     Iterator<JsonNode> it = configArray.elements();
     while (it.hasNext()) {
       JsonNode webhookConfig = it.next();
       configReads.add(toApiRead(webhookConfig));
     }
```